src/globus_cli/commands/api.py

```diff
@@ -0,0 +1,217 @@
+    click.echo(json.dumps(res.data, indent=2, separators=(",", ": "), sort_keys=True))
+
+
+@group("api")
+def api_command() -> None:
+    """Make API calls to Globus services"""
+
+
+# note: this must be written as a separate call and not inlined into the loop body
+# this ensures that it acts as a closure over 'service_name'
+def build_command(service_name: str) -> click.Command:
+    @command(
+        service_name,
+        help=f"""\
+Make API calls to Globus {service_name.title()}
+
+The arguments are an HTTP method name and a path within the service to which the request
+should be made. The path will be joined with the known service URL.
+For example, a call of
+
+    globus api {service_name} GET /foo/bar
+
+sends a 'GET' request to '{_get_url(service_name)}foo/bar'
+""",
+    )
+    @LoginManager.requires_login(_SERVICE_MAP[service_name])
+    @click.argument(
+        "method",
+        type=click.Choice(
+            ("HEAD", "GET", "PUT", "POST", "PATCH", "DELETE"), case_sensitive=False
+        ),
+    )
+    @click.argument("path")
+    @click.option(
+        "--query-param",
+        "-Q",
+        type=QueryParamType(),
+        multiple=True,
+        help="A query parameter, given as 'key=value'. Use this option multiple "
+        "times to pass multiple query parameters.",
+    )
+    @click.option(
+        "--content-type",
+        type=click.Choice(("json", "form", "text", "none", "auto")),
+        default="auto",
+        help="Use a specific Content-Type header for the request. "
+        "The default (auto) detects a content type from the data being included in "
+        "the request body, while the other names refer to common data encodings. "
+        "Any explicit Content-Type header set via '--header' will override this",
+    )
+    @click.option(
+        "--header",
+        "-H",
+        type=HeaderParamType(),
+        multiple=True,
+        help="A header, specified as 'Key: Value'. Use this option multiple "
+        "times to pass multiple headers.",
+    )
+    @click.option("--body", help="A request body to include, as text")
+    @click.option(
+        "--body-file",
+        type=click.File("r"),
+        help="A request body to include, as a file. Mutually exclusive with --body",
+    )
+    @mutex_option_group("--body", "--body-file")
+    def service_command(
+        *,
+        login_manager: LoginManager,
+        method: str,
+        path: str,
+        query_param: List[Tuple[str, str]],
+        header: List[Tuple[str, str]],
+        body: Optional[str],
+        body_file: Optional[TextIO],
+        content_type: str,
+    ):
+        client = _get_client(login_manager, service_name)
+        client.app_name = version.app_name + " raw-api-command"
+
+        query_params_d = {}
+        for param_name, param_value in query_param:
+            query_params_d[param_name] = param_value
+
+        if body_file:
+            body = body_file.read()
```

Comment:
TODO: add a comment that `body is None` is okay and desired if neither option is given